legacy/jacorb/src/main/java/org/jboss/as/jacorb/JacORBSubsystemAdd.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.jacorb;
+
+import java.util.Properties;
+
+import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathElement;
+import org.jboss.as.controller.ProcessType;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.jacorb.logging.JacORBLogger;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.Property;
+import org.wildfly.iiop.openjdk.Constants;
+import org.wildfly.iiop.openjdk.IIOPSubsystemAdd;
+
+import com.sun.corba.se.impl.orbutil.ORBConstants;
+
+/**
+ *
+ * @author <a href="mailto:tadamski@redhat.com">Tomasz Adamski</a>
+ */
+
+public class JacORBSubsystemAdd extends IIOPSubsystemAdd {
+
+    static final JacORBSubsystemAdd INSTANCE = new JacORBSubsystemAdd();
+
+    @Override
+    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model)
+            throws OperationFailedException {
+        printJacORBEmulationWarningMessage();
+        super.performRuntime(context, operation, model);
+    }
+
+    private void printJacORBEmulationWarningMessage() {
+        JacORBLogger.ROOT_LOGGER
+                .warn("Wildfly does not use JacORB as an ORB implementation anymore. JacORB subsystem would be emulated using "
+                        + "OpenJDK ORB - implementation currently used by Wildfly. Ability to emulate legacy JacORB configurations using "
+                        + "OpenJDK ORB will be removed in future.");
+    }
+
+    @Override
+    protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource)
+            throws OperationFailedException {
+        final ModelNode model = resource.getModel();
+        for (AttributeDefinition attr : JacORBSubsystemDefinitions.SUBSYSTEM_ATTRIBUTES) {
+            attr.validateAndSet(operation, model);
```

Comment:
you could avoid this for loop by just calling super(JacORBSubsystemDefinitions.SUBSYSTEM_ATTRIBUTES) in constructor.